redis/src/main/scala/zio/redis/Output.scala

```diff
@@ -385,6 +385,152 @@ object Output {
       }
   }
 
+  case object StreamInfoFullOutput extends Output[StreamInfoWithFull.FullStreamInfo] {
+    override protected def tryDecode(respValue: RespValue): StreamInfoWithFull.FullStreamInfo = {
+      var streamInfoFull: StreamInfoWithFull.FullStreamInfo = StreamInfoWithFull.FullStreamInfo.empty
+      respValue match {
+        // Note that you should not rely on the fields exact position. see https://redis.io/commands/xinfo
+        case RespValue.Array(elements) if elements.length % 2 == 0 =>
+          var pos = 0
+          while (pos < elements.length) {
+            (elements(pos), elements(pos + 1)) match {
+              // Get the basic information of the outermost stream.
+              case (key @ RespValue.BulkString(_), value @ RespValue.Integer(_)) =>
+                if (key.asString == XInfoFields.Length) streamInfoFull = streamInfoFull.copy(length = value.value)
+                else if (key.asString == XInfoFields.RadixTreeNodes)
+                  streamInfoFull = streamInfoFull.copy(radixTreeNodes = value.value)
+                else if (key.asString == XInfoFields.RadixTreeKeys)
+                  streamInfoFull = streamInfoFull.copy(radixTreeKeys = value.value)
+              case (key @ RespValue.BulkString(_), value @ RespValue.BulkString(_))
+                  if key.asString == XInfoFields.LastGeneratedId =>
+                streamInfoFull = streamInfoFull.copy(lastGeneratedId = value.asString)
+              case (key @ RespValue.BulkString(_), RespValue.Array(values)) if key.asString == XInfoFields.Entries =>
+                val streamEntryList = values.map(extractStreamEntry)
+                streamInfoFull = streamInfoFull.copy(entries = streamEntryList)
+              case (key @ RespValue.BulkString(_), RespValue.Array(value)) if key.asString == XInfoFields.Groups =>
+                // Get the group list of the stream.
+                streamInfoFull = streamInfoFull.copy(groups = extractXInfoFullGroups(value))
```

Comment:
I suggest that you create the `extractXInfoFullGroup` function and use `values.map` here, just like you did for an `extractStreamEntry`.

Reply:
It's going to be more general?

Reply:
Yes, and the implementation will be cleaner and more consistent because your function will have the extraction logic for only one value.

Reply:
done